db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1906,14 +1906,15 @@ Status DBImpl::RunManualCompaction(
       assert(!exclusive || !manual_conflict);
       // Running either this or some other manual compaction
       bg_cv_.Wait();
-      if (manual_compaction_paused_ > 0 && !manual.done &&
-          !manual.in_progress) {
+      if (manual_compaction_paused_ > 0) {
         manual.done = true;
         manual.status =
             Status::Incomplete(Status::SubCode::kManualCompactionPaused);
```

Comment:
If a compaction has been scheduled and is currently executing, should we still wait for `!manual.in_progress`?

Reply:
Thanks for pointing out. but just checking `manual.in_progress` might not enough, because the BG thread might already picked the compaction task but not yet update the `manual.in_progress` (as the thread-pool queue is not protected by db.mutex, I just added an unittest for such case).
How about make `manual` a shared_ptr?

(other options I considered but has their problems:
1. `env_->UnSchedule()` returns the number of tasks it cancelled, we can check that to make sure a scheduled compaction is cancelled, otherwise wait. The problem is it cannot work for multiple non-exclusive manual compactions as one manual compaction will unschedule() 2 jobs, another won't unschedule() any.
2. add manual job id in the Tag (hacky and may have problem if there're too many manual compactions)
3. coordinate between all FG manual compaction jobs to make sure all tasks are unscheduled, if not, wait. (I think it's possible, but seems more complicated than just make `manual` a shared_ptr.)